app/main.py

```diff
@@ -1 +1,41 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.add_alive()
+
+    def add_alive(self) -> None:
+        if self.health > 0 and self not in self.__class__.alive:
+            self.__class__.alive.append(self)
+        elif self.health <= 0 and self in self.__class__.alive:
+            self.__class__.alive.remove(self)
```

Comment:
Code Efficiency: In the `add_alive` method, you are using instance (self) to access the class attribute `alive`. It's better to use the class name to get access to the class attributes. This is a more explicit way of showing that the attribute belongs to the class, not the instance. Please, change `self.__class__.alive` to `Animal.alive`.